keps/sig-node/693-topology-manager/README.md

```diff
@@ -594,49 +593,313 @@ feature gate and set the CPU Manager policy to 'static'.
+The goal is to ensure that we don't accept enhancements with inadequate testing.
 
-* Testing the Topology Manager in a continuous integration environment
-  depends on cloud infrastructure to expose multi-node topologies
-  to guest virtual machines.
-* Implementing the `HintProvider` interface may prove challenging.
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
 
-# Limitations
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
 
-* The maximum number of NUMA nodes that Topology Manager will allow is 8,
-  past this there will be a state explosion when trying to enumerate the
-  possible NUMA affinities and generating their hints.
-* The scheduler is not topology-aware, so it is possible to be scheduled
-  on a node and then fail on the node due to the Topology Manager.
-  
-# Alternatives
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/cm/topologymanager`: `20230116` - `92.6%`
+
+##### Integration tests
+
+Not Applicable.
+
+##### e2e tests
+
+Device Manager and Device plugin node e2e tests:
+* https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/topology_manager_test.go
+
+
+### Graduation Criteria
+
+#### Alpha (v1.16) [COMPLETED]
+
+- Feature gate is disabled by default.
+- Alpha-level documentation.
+- Unit test coverage.
+- CPU Manager allocation policy takes topology hints into account.
+- Device plugin interface includes NUMA Node ID.
+- Device Manager allocation policy takes topology hints into account
+
+#### Alpha (v1.17) [COMPLETED]
+
+- Allow pods in all QoS classes to request aligned resources.
+
+#### Beta (v1.18) [COMPLETED]
+
+- Enable the feature gate by default.
+- Provide beta-level documentation.
+- Add node E2E tests.
+- Additional tests are in Testgrid and linked in KEP
+- Guarantee aligned resources for multiple containers in a pod.
+- Refactor to easily support different merge strategies for different policies.
+
+#### Beta (v1.20) [COMPLETED]
+
+* Support pod-level resource alignment.
 
-* [AutoNUMA][numa-challenges]: This kernel feature affects memory
-  allocation and thread scheduling, but does not address device locality.
+#### GA (stable) [COMPLETED]
 
-# References
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing.
+- Allowing time for user feedback.
+- Add support for device-specific topology constraints beyond NUMA.
```

Comment:
Is this a new development that blocks the GA? Same for hugepages

Reply:
IIRC there was work done to handle topology alignment at socket level in addition to just NUMA. I will try to dig up that information.

Reply:
Hugepage alignment is supported now with the introduction of memory manager.

Reply:
sorry for confusion, thank you!